packages/combo-box/src/vaadin-combo-box-data-provider-mixin.js

```diff
@@ -177,40 +177,42 @@ export const ComboBoxDataProviderMixin = (superClass) =>
     /** @private */
     _loadPage(page) {
       // Make sure same page isn't requested multiple times.
-      if (!this._pendingRequests[page] && this.dataProvider) {
-        this.loading = true;
-
-        const params = {
-          page,
-          pageSize: this.pageSize,
-          filter: this.filter,
-        };
-
-        const callback = (items, size) => {
-          if (this._pendingRequests[page] === callback) {
-            const filteredItems = this.filteredItems ? [...this.filteredItems] : [];
-            filteredItems.splice(params.page * params.pageSize, items.length, ...items);
-            this.filteredItems = filteredItems;
-
-            if (!this.opened && !this.hasAttribute('focused')) {
-              this._commitValue();
-            }
-            this.size = size;
-
-            delete this._pendingRequests[page];
-
-            if (Object.keys(this._pendingRequests).length === 0) {
-              this.loading = false;
-            }
-          }
-        };
+      if (this._pendingRequests[page] || !this.dataProvider) {
+        return;
+      }
+
+      const params = {
+        page,
+        pageSize: this.pageSize,
+        filter: this.filter,
+      };
 
-        if (!this._pendingRequests[page]) {
```

Comment:
**note:** This check is unnecessary because the same check is done at the beginning of the method.